### PR TITLE
feat(dgw): add endpoint for network interface config

### PR DESCRIPTION
**Pull request by irvingoujAtDevolution**
waiting for #698 to be merged first.

Added api endpoint for ipconfig, reuse network token type. It is not necessary to create a new token type specifically for this.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -13,4 +13,8 @@
     "powershell.codeFormatting.whitespaceBeforeOpenBrace": true,
     "powershell.codeFormatting.whitespaceBeforeOpenParen": true,
     "powershell.codeFormatting.whitespaceBetweenParameters": true,
+    "rust-analyzer.linkedProjects": [
+        ".\\devolutions-gateway\\Cargo.toml",
+        ".\\devolutions-gateway\\Cargo.toml"
+    ],
```
Comment (CBenoit) [inline .vscode/settings.json:19]: issue: Leftover?

**Review by CBenoit — COMMENTED** [review]
Can you rebase on master so I can review without the duplicate changes? Thank you!

File: devolutions-gateway/src/api/network_scan/mod.rs [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+use axum::Router;
+
+pub mod ipconfig;
+pub mod scan;
+
+pub fn make_router<S>(state: DgwState) -> Router<S> {
+    Router::new()
+        .route("/", axum::routing::get(scan::handler))
+        .route("/ipconfig", axum::routing::get(ipconfig::handler))
+        .with_state(state)
```
Comment (CBenoit) [inline devolutions-gateway/src/api/network_scan/mod.rs:11]: suggestion: If we’re going to have nested routes like this, I would suggest something like this:

- `/jet/net/scan`: Run the network scanning (WebSocket)
- `/jet/net/config`: Fetch the network interfaces configuration (plain HTTP call)

Comment (CBenoit) [reply devolutions-gateway/src/api/network_scan/mod.rs:11]: See https://github.com/Devolutions/devolutions-gateway/pull/709

File: .vscode/settings.json (hunk 1 above)
Comment (irvingoujAtDevolution) [reply .vscode/settings.json]: updated

File: devolutions-gateway/src/api/network_scan/mod.rs (hunk 2 above)
Comment (irvingoujAtDevolution) [reply devolutions-gateway/src/api/network_scan/mod.rs]: updated

File: devolutions-gateway/src/api/network_scan/mod.rs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+use crate::DgwState;
+use axum::Router;
+
+pub mod ipconfig;
+pub mod scan;
+
+pub fn make_router<S>(state: DgwState) -> Router<S> {
+    Router::new()
+        .route("/scan", axum::routing::get(scan::handler))
+        .route("/ipconfig", axum::routing::get(ipconfig::handler))
```
Comment (CBenoit) [inline devolutions-gateway/src/api/network_scan/mod.rs:10]: issue: The endpoint should be `/jet/net/config`
```suggestion
        .route("/config", axum::routing::get(ipconfig::handler))
```

File: devolutions-gateway/src/api/network_scan/ipconfig.rs [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+use network_scanner::interfaces::{MacAddr, NetworkInterface};
+use serde::Serialize;
+
+use crate::http::HttpError;
+
+pub async fn handler(_netscan_claim: crate::extract::NetScanToken) -> Result<Response, HttpError> {
+    let res = network_scanner::interfaces::get_network_interfaces().map_err(|e| {
+        tracing::error!("failed to get network interfaces: {:?}", e);
+        HttpError::internal().build(e)
+    })?;
```
Comment (CBenoit) [inline devolutions-gateway/src/api/network_scan/ipconfig.rs:14]: suggestion: This is how it’s supposed to be used. The error will be logged automatically.
```suggestion
    let res = network_scanner::interfaces::get_network_interfaces()
        .map_err(HttpError::internal().msg("failed to get network interfaces"))?;
```

**Review by CBenoit — APPROVED** [review]
I’ll merge now to smoothen the process.

**Pull request merged**